.github/workflows/build-deploy.yml

```diff
@@ -26,74 +30,147 @@ jobs:
     steps:
-      - name: Get GHA environment name
-        id: env_name
+      - uses: styfle/cancel-workflow-action@0.11.0
+        with:
+          access_token: ${{ github.token }}
+      - id: configure
         run: |
-          echo "Running on branch ${{ github.ref }}"
-          if [ "${{ github.ref }}" = "refs/heads/develop" ]; then
-            echo "::set-output name=env_name::dev"
-          elif [ "${{ github.ref }}" = "refs/heads/master" ]; then
-            echo "::set-output name=env_name::staging"
-          elif [ "${{ github.ref }}" = "refs/heads/release" ]; then
-            echo "::set-output name=env_name::prod"
-          fi
-      - name: Use specified environment secrets from previous step
-        run: echo "Environment secrets set name ${{ steps.env_name.outputs.env_name }}"
+              echo "Running on branch ${{ github.ref }}"
+              if [ "${{ github.ref }}" = "refs/heads/develop" ]; then
+                echo "ENVIRONMENT=dev" >> $GITHUB_OUTPUT
+                echo "PRERELEASE=true" >> $GITHUB_OUTPUT
+                echo "IMAGE_TAG=$(uuidgen)" >> $GITHUB_OUTPUT
+                echo "TAG_DRY_RUN=true" >> $GITHUB_OUTPUT
+                echo "PUSH_DOCKER=false" >> $GITHUB_OUTPUT
+              elif [ "${{ github.ref }}" = "refs/heads/master" ]; then
+                echo "ENVIRONMENT=staging" >> $GITHUB_OUTPUT
+                echo "PRERELEASE=true" >> $GITHUB_OUTPUT
+                echo "IMAGE_TAG=staging" >> $GITHUB_OUTPUT
+                echo "TAG_DRY_RUN=false" >> $GITHUB_OUTPUT
+                echo "PUSH_DOCKER=true" >> $GITHUB_OUTPUT
+              elif [ "${{ github.ref }}" = "refs/heads/release" ]; then
+                echo "ENVIRONMENT=prod" >> $GITHUB_OUTPUT
+                echo "PRERELEASE=false" >> $GITHUB_OUTPUT
+                echo "IMAGE_TAG=latest" >> $GITHUB_OUTPUT
+                echo "TAG_DRY_RUN=false" >> $GITHUB_OUTPUT
+                echo "PUSH_DOCKER=true" >> $GITHUB_OUTPUT
+              else
+                echo "Branch not configured"
+                exit 1
+              fi
+              
+  tag-release:
+    name: Push tag and create release
+    runs-on: ubuntu-latest
+    needs: [configure]
     outputs:
-      env_name: ${{ steps.env_name.outputs.env_name }}
+      tag: ${{ steps.tag_version.outputs.new_tag }}
+    steps:
+      - uses: actions/checkout@v3
 
-  build-push-deploy:
+      - name: Bump version and push tag
+        id: tag_version
+        uses: mathieudutour/github-tag-action@v6.1
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          release_branches: release
+          pre_release_branches: master    # these branches will produce pre-release tag (eg. v0.1.2-beta.0)
+          append_to_pre_release_tag: beta
+          custom_release_rules: major:major:Major Changes,minor:minor:Minor Changes,chore:patch:Chores
+          dry_run: ${{  needs.configure.outputs.TAG_DRY_RUN  }}   # pushes to develop won't trigger tagging
```

Comment:
This looks good, my only concern with this is that the version in `package.json` won't reflect this tag & release version. Maybe this is okay, but curious if you are aware of a way to improve this.

Reply:
good to know, thanks. I thinking you're saying that we can set `previous_tag` so that it will start from current version. I hope actually that won't be a problem as we do already have version tags.
I mean more that the version in the package.json file won't be updated: https://github.com/energywebfoundation/ssi-hub/blob/fff75267452ea4c411de560df5f837d0dc453c9e/package.json#L3
But I think that's okay, we can live with this mismatch for now and sort it out later.